### PR TITLE
Added dockerfile and environment files.

### PR DESCRIPTION
**Pull request by jejjohnson**
Preliminary steps towards docker.

* I added a preliminary `Dockerfile` that is constructed and uploaded to gcp. 
* I also created an `environment.yml` and `environment_dev.yml` files to be used for the users and dockers when creating the environments using `conda`.
* Updated the `.gitignore` file to not save the `.pkl` files and `.png` files.

Everything was tested with the `example.py` script.

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -2,4 +2,18 @@ name: dotenv
 channels:
   - anaconda
 dependencies:
-  - theano
\ No newline at end of file
+  - python=3.8
```
Comment (bmorris3) [inline environment.yml:5]: This particular change might make merging portions with master complicated at some point in the future. Would it be possible to copy the old file and rename it `rtd-environment.yml`; then edit [`.readthedocs.yml`](https://github.com/bmorris3/dot/blob/master/.readthedocs.yml#L16)` to point to rtd-environment.yml?

Comment (jejjohnson) [reply environment.yml:5]: Updated the changes you requested. I put the `rtd-environment.yml` file inside the `docs/` folder and updated the `.readthedocs.yml` to point to that location.

File: .readthedocs.yml [hunk 2]
```diff
@@ -13,6 +13,6 @@ python:
         - all
 
 conda:
-  environment: environment.yml
+  environment: docs/rtd-environment.yml/environment.yml
```
Comment (bmorris3) [inline .readthedocs.yml]: Should this be ` environment: docs/rtd-environment.yml`?

Comment (bmorris3) [reply .readthedocs.yml]: I figured out how to add a commit to your PR easily, so don't worry about changing this, I got it!

**Comment by bmorris3** [comment]
Tests passed, looks good. Thanks @jejjohnson! 🎉

**Pull request merged**